### DISCUSSION TITLE
Running API and UI services on different hosts

### DISCUSSION THREAD

**Discussion by mafrosis** [photoview/photoview#1146] [Q&A and Troubleshooting] [answered]
I use docker to run photoview currently on a rather underpowered host. I'm interested in the possibility to run the API on different server from the UI, has anyone looked into such a thing?

**Comment by kkovaletp** [comment, marked as answer]
@mafrosis, I think you can achieve this using a reverse proxy, which can serve the UI by itself, so you can disable the `PHOTOVIEW_SERVE_UI` env variable for the Photoview container.

Here is a documentation page for reverse proxy integrations: https://photoview.github.io/en/docs/installation-reverse-proxies/
In addition to the standard integration, you'd need to extract the UI files from the Photoview image (`/app/ui`) and copy them to the Reverse proxy. Then you need to configure the Reverse proxy to act as a web server and serve the UI from these files while forwarding all the API requests to the Photoview container.
